sycl/unittests/Extensions/FreeFunctionCommands/Barrier.cpp

```diff
@@ -0,0 +1,84 @@
+#include <helpers/TestKernel.hpp>
+#include <helpers/UrMock.hpp>
+
+#include <gtest/gtest.h>
+
+#define __DPCPP_ENABLE_UNFINISHED_KHR_EXTENSIONS
+#include <sycl/ext/khr/free_function_commands.hpp>
+
+static size_t NumOfEnqueueEventsWaitWithBarrier = 0;
```

Comment:
```suggestion
thread_local size_t NumOfEnqueueEventsWaitWithBarrier = 0;
```
Just in case there are configurations that run in parallel.

Reply:
Sure! Merged in https://github.com/intel/llvm/pull/16770/commits/4753bd4206d1aa564a1bb246896fbe4ea2b16bbf